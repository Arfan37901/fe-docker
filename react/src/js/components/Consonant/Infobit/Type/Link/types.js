import {
    bool,
    string,
    oneOfType,
} from 'prop-types';

// eslint-disable-next-line import/prefer-default-export
export const LinkType = {
    linkHint: string,
    href: string.isRequired,
    text: string.isRequired,
    openInNewTab: oneOfType([bool, string]),
};
