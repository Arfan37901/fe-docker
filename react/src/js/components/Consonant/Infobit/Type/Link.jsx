import React from 'react';
// import PropTypes from 'prop-types';

function Link() {
    return (<span>Link bit</span>);
}

export default Link;
