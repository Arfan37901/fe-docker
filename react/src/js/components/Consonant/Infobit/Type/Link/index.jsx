import React from 'react';

import { LinkType } from './types';

const defaultProps = {
    linkHint: '',
    openInNewTab: true,
};

/**
 * Link Infobit (shown in 3:2 Card Footer)
 *
 * @component
 * @example
 * const props= {
    href: String,
    openInNewTab: Boolean,
    linkHint: String,
    text: String,
 * }
 * return (
 *   <LinkWithIcon {...props}/>
 * )
 */
const Link = ({
    href,
    openInNewTab,
    linkHint,
    text,
}) => {
    const target = openInNewTab ? '_blank' : '_self';
    return (
        <a
            className="consonant-link-infobit"
            data-testid="link-infobit"
            href={href}
            target={target}
            title={linkHint}
            rel="noopener noreferrer"
            tabIndex="0">
            {text}
        </a>
    );
};

Link.propTypes = LinkType;
Link.defaultProps = defaultProps;

export default Link;
