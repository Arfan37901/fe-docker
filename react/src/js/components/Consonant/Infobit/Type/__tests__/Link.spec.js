import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import Link from '../Link';

const props = {
    href: '/some/valid/uri',
    openInNewTab: false,
    text: 'text',
};

describe('Consonant/Bookmarks', () => {
    test('should correct render', async () => {
        const tree = renderer
            .create(<Link {...props} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
    test('should correct render with current tab', async () => {
        const tree = renderer
            .create(<Link {...props} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
