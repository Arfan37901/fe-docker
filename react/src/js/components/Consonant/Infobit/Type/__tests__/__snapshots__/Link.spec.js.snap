// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/Bookmarks should correct render 1`] = `
<a
  className="consonant-link-infobit"
  href="/some/valid/uri"
  rel="noreferrer"
  target="_self"
  title=""
>
  text
</a>
`;

exports[`Consonant/Bookmarks should correct render with current tab 1`] = `
<a
  className="consonant-link-infobit"
  href="/some/valid/uri"
  rel="noreferrer"
  target="_self"
  title=""
>
  text
</a>
`;
